### PR TITLE
feat: Add disconnect button on auth config pages and form login callout banner

### LINKED ISSUE #9936: Configuration Page for authentication options

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Earlier, for disconnecting any authentication method viz. Google/Github, the super admin had to empty each of the fields and click Save & Restart button. This has been fixed in this PR by the addition of a disconnect button on each of the authentication method config pages.
> Added a form login callout banner on the form login auth config page, which warns the super admin that the email IDs used in form login/signup will not be verified.

Fixes #9936

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

> Disconnect button
- Added config needed for enabling google authentication.
- Once it is enabled, checked if disconnect button is seen on google authentication config page.
- Clicked on disconnect button and checked if Google authentication gets disabled on login screen.

> Form login callout banner
- Went to form login authentication config page.
- Checked if a red warning banner exists with a link to docs.appsmith.com

Screen recording of changes done on local machine:
[LOOM DEMO](https://www.loom.com/share/12e62be2341d4e9f87198f2d3d1a37bf)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: feat/form-login-calout 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.67 **(-0.07)** | 36.99 **(-0.09)** | 35.15 **(-0.67)** | 56.01 **(-0.09)**
 :red_circle: | app/client/src/ce/configs/index.ts | 97.3 **(0)** | 86.14 **(-1.49)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/ce/constants/messages.ts | 76.93 **(-0.05)** | 100 **(0)** | 31.6 **(-0.15)** | 81.72 **(-0.22)**
 :sparkles: :new: | **app/client/src/components/ads/ColorSelector.tsx** | **21.43** | **22.22** | **0** | **23.08**
 :sparkles: :new: | **app/client/src/components/ads/DisplayImageUpload.tsx** | **28.57** | **12.5** | **0** | **32**
 :sparkles: :new: | **app/client/src/components/ads/IconSelector.tsx** | **22.73** | **8.89** | **0** | **25.64**
 :sparkles: :new: | **app/client/src/components/ads/NumberedStep.tsx** | **54.55** | **100** | **0** | **54.55**
 :sparkles: :new: | **app/client/src/components/ads/RectangularSwitcher.tsx** | **26.67** | **30** | **0** | **30.77**
 :sparkles: :new: | **app/client/src/components/ads/SearchInput.tsx** | **42.86** | **21.43** | **14.29** | **45.45**
 :sparkles: :new: | **app/client/src/components/ads/ShowcaseCarousel.tsx** | **26.53** | **37.5** | **0** | **28.26**
 :sparkles: :new: | **app/client/src/components/ads/Statusbar.tsx** | **45** | **44.44** | **0** | **45**
 :sparkles: :new: | **app/client/src/components/ads/TabItemBackgroundFill.tsx** | **46.15** | **50** | **0** | **50**
 :sparkles: :new: | **app/client/src/components/ads/Table.tsx** | **19.15** | **50** | **0** | **19.15**
 :sparkles: :new: | **app/client/src/components/ads/TableDropdown.tsx** | **27.66** | **28.57** | **0** | **31.71**
 :red_circle: | app/client/src/components/ads/TagInputComponent.tsx | 13.11 **(0)** | 4.08 **(-1.05)** | 0 **(0)** | 14.04 **(0)**
 :sparkles: :new: | **app/client/src/components/ads/UserProfileImagePicker.tsx** | **23.33** | **0** | **0** | **25**
 :sparkles: :new: | **app/client/src/components/ads/index.ts** | **100** | **100** | **0** | **100**
 :sparkles: :new: | **app/client/src/components/ads/formFields/RedirectUrlForm.tsx** | **72** | **100** | **0** | **72**
 :sparkles: :new: | **app/client/src/components/ads/formFields/UneditableField.tsx** | **76.92** | **50** | **0** | **76.92**
 :green_circle: | app/client/src/constants/ThirdPartyConstants.tsx | 91.67 **(0.76)** | 100 **(0)** | 0 **(0)** | 90.91 **(0.91)**
 :sparkles: :new: | **app/client/src/pages/Settings/FormGroup/Accordion.tsx** | **50** | **57.14** | **0** | **55**
 :sparkles: :new: | **app/client/src/pages/Settings/FormGroup/TagInputField.tsx** | **66.67** | **0** | **0** | **66.67**
 :red_circle: | app/client/src/pages/Settings/FormGroup/TextInput.tsx | 100 **(0)** | 60 **(-15)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/pages/Settings/FormGroup/group.tsx | 91.11 **(-6.19)** | 50.75 **(-21.47)** | 100 **(0)** | 90.91 **(-6.31)**</details>

### CODE REVIEW

File: app/client/src/ce/pages/AdminSettings/config/authentication/index.tsx [hunk 1]
```diff
@@ -51,12 +53,22 @@ const Form_Auth: AdminConfigType = {
...
       toggleText: (value: boolean) => {
         if (value) {
-          return "Allow invited users to signup";
+          return "Allow only invited users to signup";
         } else {
           return " Allow all users to signup";
         }
       },
     },
+    {
```
Comment (pranavkanade) [inline app/client/src/ce/pages/AdminSettings/config/authentication/index.tsx:62]: Do we need this in the final version? As I understand, we do not have any condition to show this banner.

File: app/client/src/pages/Settings/FormGroup/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
```
Comment (pranavkanade) [inline app/client/src/pages/Settings/FormGroup/Accordion.tsx:1]: Can we move this component to ADS?

File: app/client/src/ce/pages/AdminSettings/config/types.ts [hunk 3]
```diff
@@ -10,6 +10,8 @@ export enum SettingTypes {
   GROUP = "GROUP",
   TEXT = "TEXT",
   PAGE = "PAGE",
+  REDIRECT = "REDIRECT",
```
Comment (pranavkanade) [inline app/client/src/ce/pages/AdminSettings/config/types.ts:13]: I suppose this is for showing Redirect URL? If yes, can we not make it generic? Like - Uneditable Field? In my understanding, this file should have generic types.

File: deploy/template/docker.env.sh [hunk 4]
```diff
@@ -35,6 +35,11 @@ APPSMITH_MAIL_ENABLED=false
 # APPSMITH_OAUTH2_GITHUB_CLIENT_SECRET=
 # *********************************
 
+# ******** Form Login/Signup ********
```
Comment (pranavkanade) [inline deploy/template/docker.env.sh:38]: This is a duplicate block.

File: app/client/src/pages/Settings/FormGroup/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const AccordionBody = styled.div`
+  & .hide {
+    display: none;
+  }
+  & .callout-link {
+    max-width: 634px;
```
Comment (albinAppsmith) [inline app/client/src/pages/Settings/FormGroup/Accordion.tsx:31]: nit: can we make this px to relative units like rem/vw

File: app/client/src/pages/Settings/FormGroup/RedirectUrlForm.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  width: 100%;
+  padding: 0 0 16px;
+`;
+
+const HeaderWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 8px;
+  .help-icon {
+    margin-left: 5px;
```
Comment (albinAppsmith) [inline app/client/src/pages/Settings/FormGroup/RedirectUrlForm.tsx:27]: nit: we should be using multiples of 4 for units in px

File: app/client/src/pages/Settings/SettingsForm.tsx [hunk 7]
```diff
@@ -28,7 +44,9 @@ const Wrapper = styled.div`
   overflow: auto;
 `;
 
-const SettingsFormWrapper = styled.div``;
+const SettingsFormWrapper = styled.div`
+  max-width: 634px;
```
Comment (albinAppsmith) [inline app/client/src/pages/Settings/SettingsForm.tsx:48]: nit: relative units

**Review by albinAppsmith — APPROVED** [review]

File: app/client/src/components/ads/formFields/UneditableField.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+} from "redux-form";
+import InputComponent, { InputType } from "../TextInput";
+import { Intent } from "constants/DefaultTheme";
+import styled from "styled-components";
+import { ReactComponent as CopyIcon } from "assets/icons/menu/copy-snippet.svg";
+
+const Label = styled.div`
+  font-size: 14px;
+  margin: 8px 0;
+  color: #4b4848;
```
Comment (albinAppsmith) [inline app/client/src/components/ads/formFields/UneditableField.tsx:15]: use colors from Colors constant

File: app/client/src/pages/Settings/SettingsForm.tsx [hunk 9]
```diff
@@ -136,6 +182,15 @@ export function SettingsForm(
             valid={props.valid}
           />
         )}
+        {details?.isConnected && (
+          <DisconnectService
+            disconnect={() => disconnect(settings)}
```
Comment (pranavkanade) [inline app/client/src/pages/Settings/SettingsForm.tsx:187]: From the video, I understand that there is no secondary prompt when disconnecting. Can you please ask Joy to take a look once?

File: app/client/src/pages/Settings/SettingsForm.tsx [hunk 10]
```diff
@@ -79,9 +97,20 @@ export function SettingsForm(
...
     subCategory ?? category,
   );
+  const pageTitle = getSettingLabel(
+    details?.title || (subCategory ?? category),
+  );
 
   const onSave = () => {
-    dispatch(saveSettings(props.settings));
+    if (
+      ("APPSMITH_FORM_LOGIN_DISABLED" in props.settings &&
```
Comment (pranavkanade) [inline app/client/src/pages/Settings/SettingsForm.tsx:106]: I think you should use similar logic here which you've used at the bottom -
```
    const connectedMethods = [
      enableGoogleOAuth,
      enableGithubOAuth,
      !disableLoginForm,
    ].filter(Boolean);
    if (connectedMethods.length >= 2) {
```

File: app/client/src/pages/Settings/FormGroup/RedirectUrlForm.tsx [hunk 11]
```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from "react";
```
Comment (pranavkanade) [inline app/client/src/pages/Settings/FormGroup/RedirectUrlForm.tsx:1]: Can we move this component to ADS?

**Review by pranavkanade — DISMISSED** [review]
Few suggestions.

**Pull request merged**